### PR TITLE
Better installation errors #2

### PR DESCRIPTION
**Pull request by davdebcom**
Tip found at https://www.rsjoomla.com/support/view-article/1012-troubleshooting-extension-installation-errors.html

Example of what patch does:
image

### CODE REVIEW

**Comment by wilsonge** [comment]
Hi,
Thanks for introducing this improvement to the CMS!

Please use a language string rather than hardcoding strings so that they are translatable. Also as this will change language strings it will have to wait until after 3.4 release (as we are currently in a language freeze)

**Comment by brianteeman** [comment]
Is it not possible that this could be committed as a special case with the hardcoded strings NOW and then updated to use jtext later?

**Comment by Bakual** [comment]
Imho it's not urgent enough to make a special exception for that.

**Comment by sovainfo** [comment]
Suggest to use:
JText::sprintf('JLIB_FILESYSTEM_ERROR_WARNFS_ERR02' . ' : %s => %s' , $src, $dest)

Could be changed later if required

**Comment by davdebcom** [comment]
@sovainfo Changed to match your suggestion

Ready to go?

**Comment by sovainfo** [comment]
Removed the space before the comma to make travis happy, sorry. Made a PR against yours.

What about the other JText::_ calls?
Like line 145 JLIB_FILESYSTEM_ERROR_COPY_FAILED

**Comment by davdebcom** [comment]
@sovainfo Merged into my patch, thanks!

**Comment by designbengel** [comment]
Hi Guys, how to test this ? thx<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6002">issues.joomla.org/joomla-cms/6002</a>.</sub>

**Comment by wilsonge** [comment]
From what I understand you need to install an extension when having not enough file permissions in the components (or some folder that's needed as described in the link in the description) to move files into the directory. Before the patch you'll just see a message "Copy file failed" and after you'll see a message stating what the to and from directory are.

**Comment by designbengel** [comment]
Thanks, - tested successfully - just the translations are missing 
JLIB_FILESYSTEM_ERROR_WARNFS_ERR02 : /Applications/MAMP/tmp/php/phpLYx2kB => /Applications/MAMP/htdocs/joomla-cms-staging/tmp/com_patchtester.tar.bz2

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6002">issues.joomla.org/joomla-cms/6002</a>.</sub>

**Comment by zero-24** [comment]
RTC based on testing. Thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6002">issues.joomla.org/joomla-cms/6002</a>.</sub>

File: libraries/joomla/filesystem/file.php [hunk 1]
```diff
@@ -140,7 +140,7 @@ public static function copy($src, $dest, $path = null, $use_streams = false)
 			{
 				if (!@ copy($src, $dest))
 				{
-					JLog::add(JText::_('JLIB_FILESYSTEM_ERROR_COPY_FAILED'), JLog::WARNING, 'jerror');
+					JLog::add(JText::_('JLIB_FILESYSTEM_ERROR_COPY_FAILED' . ' : %s => %s', $src, $dest), JLog::WARNING, 'jerror');
```
Comment (phproberto) [inline libraries/joomla/filesystem/file.php]: This is not using `JText::sprintf` so it won't work as expected.

**Comment by phproberto** [comment]
Still requires a fix.

**Comment by zero-24** [comment]
Moving back to Pending based on @phproberto's commet.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6002">issues.joomla.org/joomla-cms/6002</a>.</sub>

**Comment by sovainfo** [comment]
@davdebcom created pr to add sprintf to line 143

**Comment by infograf768** [comment]
This does not work! : the constant is not translated.

the code should be:
```
diff --git a/libraries/joomla/filesystem/file.php b/libraries/joomla/filesystem/file.php
index cc14b9c..585be6a 100644
--- a/libraries/joomla/filesystem/file.php
+++ b/libraries/joomla/filesystem/file.php
@@ -141,5 +141,5 @@
 				if (!@ copy($src, $dest))
 				{
-					JLog::add(JText::_('JLIB_FILESYSTEM_ERROR_COPY_FAILED'), JLog::WARNING, 'jerror');
+					JLog::add(JText::sprintf(JText::_('JLIB_FILESYSTEM_ERROR_COPY_FAILED') . ' : %1$s => %2$s', $src, $dest), JLog::WARNING, 'jerror');
 
 					return false;
@@ -520,5 +520,5 @@
 				else
 				{
-					JLog::add(JText::_('JLIB_FILESYSTEM_ERROR_WARNFS_ERR02'), JLog::WARNING, 'jerror');
+					JLog::add(JText::sprintf(JText::_('JLIB_FILESYSTEM_ERROR_WARNFS_ERR02') . ' : %1$s => %2$s', $src, $dest), JLog::WARNING, 'jerror');
 				}
 			}
@@ -539,5 +539,5 @@
 				else
 				{
-					JLog::add(JText::_('JLIB_FILESYSTEM_ERROR_WARNFS_ERR02'), JLog::WARNING, 'jerror');
+					JLog::add(JText::sprintf(JText::_('JLIB_FILESYSTEM_ERROR_WARNFS_ERR02') . ' : %1$s => %2$s', $src, $dest), JLog::WARNING, 'jerror');
 				}
 			}
 $src, $dest), JLog::WARNING, 'jerror');
```

Evidently, as the variables and the hardcoding of `=>` are not in the language string value, we will get weird results for a RTL Language:

screen shot 2015-04-02 at 10 52 15

If we want to also solve this we could do it this way:
```
if (JFactory::getLanguage()->isRTL())
{
JLog::add(JText::sprintf(JText::_('JLIB_FILESYSTEM_ERROR_WARNFS_ERR02') . ' : %2$s <= %1$s', $src, $dest), JLog::WARNING, 'jerror');
}
else
{
JLog::add(JText::sprintf(JText::_('JLIB_FILESYSTEM_ERROR_WARNFS_ERR02') . ' : %1$s => %2$s', $src, $dest), JLog::WARNING, 'jerror');
}
```
we would get:
screen shot 2015-04-02 at 11 08 21

**Comment by sovainfo** [comment]
Added PR fixing translation and ltr by changing the messages in the language files.

**Comment by infograf768** [comment]
@sovainfo 
I commented on your PR. I suggest you link it to this and explain there how to test.

**Comment by sovainfo** [comment]
Changed my PR according to instructions from @infograf768. @davdebcom needs to accept my PR if he agrees.

**Pull request closed without merging**